### ISSUE TITLE
No parameter named "is_dst" error from package pytz

### ISSUE THREAD

**Issue by bschnurr** [microsoft/pyright#3364]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**

Getting a error when trying to use the named argument `is_dst` in the `pactz.ucoffset(..)` function 

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.

pip install pytz

```python
import pytz
from datetime import datetime

dt = datetime(2019, 10, 31, 23, 30)
pactz.utcoffset(dt, is_dst=True) 
```

errors, even though `dt` is class datetime
![image](https://user-images.githubusercontent.com/1946977/164340149-175bfc4a-7ba1-4d0d-a1ba-3dfca8725661.png)

typeshed stub 
![image](https://user-images.githubusercontent.com/1946977/164339813-adc4f4a5-1a76-41f5-8585-5fb338a63ad4.png)

not sure how it knows the first overload in the list has a `dt` of type `datetime` because the first overload in typeshed has `dt` as `None`
![image](https://user-images.githubusercontent.com/1946977/164340457-b8b8a0e7-c8b8-41e9-baea-9dc9932e9a64.png)

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

**Additional context**

**Comment by erictraut** [comment]
Can you fix the code sample? The `pactz` symbol isn't defined.

**Comment by erictraut** [comment]
Oh, never mind, I see that it's correct in the screen shot.

```python
import pytz

pactz = pytz.timezone('')
pactz.utcoffset(dt, is_dst=True) 
```

**Comment by erictraut** [comment]
Pyright is correct in emitting an error here. The function `pytz.timezone` [returns a union of three types](https://github.com/python/typeshed/blob/2116d2425e5953b4944a14450cb451947adfadab/stubs/pytz/pytz/__init__.pyi#L23): `_UTCclass | StaticTzInfo | DstTzInfo`. It is not overloaded, so those types do not depend on input parameters.

All three of these types have a method called `utcoffset`, but only two of the three (`StaticTzInfo` and `DstTzInfo`) accept a named parameter called `is_dst`. If the type of object returned by `pytz.timezone()` happens to be a `[_UTCclass](https://github.com/python/typeshed/blob/2116d2425e5953b4944a14450cb451947adfadab/stubs/pytz/pytz/__init__.pyi#L17)`, it will crash at runtime.

This appears to be a bug in the typeshed stubs. Please file a bug or a PR there. 

Looking at the source for the latest published version of pytz, it looks like none of the `utcoffset` methods accept a parameter of the name `is_dst`.

@JelleZijlstra, it looks like you recently updated the pytz stubs. Perhaps you have more context here?

**Comment by JelleZijlstra** [comment]
@erictraut what is the bug that you're suggesting exists in the stubs? I looked at the latest release of pytz and `pytc.UTC.utcoffset` indeed doesn't take an `is_dst` argument, but `StaticTzInfo.utcoffset` does.

**Comment by erictraut** [comment]
Thanks for confirming. I didn't realize that the class `_UTCclass` in the stub mapped to the class `pytz.UTC` in the implementation. I guess that means callers need to do an `isinstance(x, StaticTzInfo | DstTzInfo)` check before they can safely invoke `utcoffset` with an `is_dst` named argument.
